src/client.js

```diff
@@ -445,9 +445,10 @@ MatrixClient.prototype.setNotifTimelineSet = function(notifTimelineSet) {
  * @return {module:http-api.MatrixError} Rejects: with an error response.
  */
 MatrixClient.prototype.getCapabilities = function() {
```

Comment:
The current code seems to hit the server twice in quick succession during startup, perhaps because two code paths call this before we're able to receive and cache the first attempt.

I wonder if that's at all related to the issue here, perhaps if the server were rate limiting or something and sent an error response the second time...?

Anyway, it's likely something worth improving at some stage, though doesn't have to be this PR. (Just mentioning here in case it helps.)